dias/rewriter.py

```diff
@@ -929,30 +941,65 @@ def rewrite_ast(cell_ast: ast.Module) -> Tuple[str, Dict]:
 
       stats[type(patt).__name__] = 1
     elif isinstance(patt, patt_matcher.ReplaceRemoveList):
-      # We can't process it until we know the function passed to apply().
-      # Save it as a deferred pattern.
+      """This rewrites `e.replace([x1], [x2], **kwargs)` as:
+      dias.rewriter.replace_remove_list(e, [x1], [x2], x1, x2, 
+        lambda _DIAS_x1, _DIAS_x2: e.replace(_DIAS_x1, _DIAS_x2, **kwargs))
+      """
       assert len(stmt_idxs) == 1
       stmt_idx = stmt_idxs[0]
       stmt_list = list_of_lists[stmt_idx]
+
+      df_or_ser = patt.attr_call.get_called_on()
+
+      call_raw = patt.attr_call.call.get_obj()      
+      orig_to_replace = call_raw.args[0]
+      orig_replace_with = call_raw.args[1]
+
+      call_raw.args = [AST_name(name="_DIAS_x1"), AST_name(name="_DIAS_x2")]
```

Comment:
Just to be sure. This does preserve `**kwargs` because you're only changing `.args` and the `kwargs` remain as they were. Correct?

Reply:
Yes, and the `ReplaceRemoveList` pattern matcher ensures that `args` has exactly two elements, so this preserves both `args` as well as `kwargs`. This gets tested with the `inplace=True` kwarg in the test cases.